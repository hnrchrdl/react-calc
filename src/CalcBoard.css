.calc-board {
  display: inline-flex;
  flex-direction: column;
  margin: 10px;
  padding: 10px;
  border: 1px solid #eee;
}
.calc-board-container {
  display: flex;
  margin: 0 auto;
}
