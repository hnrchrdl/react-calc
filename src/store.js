import { createStore, combineReducers } from 'redux';
import calcReducer from './calculator';

export default createStore(combineReducers({ calculator: calcReducer }));
