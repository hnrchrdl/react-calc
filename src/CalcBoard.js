import React from 'react';
import Pad from './Pad';
import './CalcBoard.css';
import Display from './Display';

const CalcBoard = () => {
  return (
    <div className="calc-board">
      <Display />
      <div className="calc-board-container">
        <div className="calc-board-rows">
          <div className="calc-board-row">
            <Pad value={1} display="1" />
            <Pad value={2} display="2" />
            <Pad value={3} display="3" />
          </div>
          <div className="calc-board-row">
            <Pad value={4} display="4" />
            <Pad value={5} display="5" />
            <Pad value={6} display="6" />
          </div>
          <div className="calc-board-row">
            <Pad value={7} display="7" />
            <Pad value={8} display="8" />
            <Pad value={9} display="9" />
          </div>
          <div className="calc-board-row">
            <Pad value={0} display="0" />
            <Pad value={'.'} display="." />
          </div>
        </div>
        <div className="calc-board-rows">
          <div className="calc-board-row">
            <Pad value={'+'} display="+" />
          </div>
          <div className="calc-board-row">
            <Pad value={'-'} display="-" />
          </div>
          <div className="calc-board-row">
            <Pad value={'*'} display="*" />
          </div>
          <div className="calc-board-row">
            <Pad value={'/'} display="/" />
          </div>
        </div>
        <div className="calc-board-rows">
          <div className="calc-board-row">
            <Pad value={'undo'} display="C" />
          </div>
          <div className="calc-board-row">
            <Pad value={'reset'} display="CC" />
          </div>
        </div>
        <div className="calc-board-rows">
          <Pad value={'ok'} display="=" />
        </div>
      </div>
    </div>
  );
};

export default CalcBoard;
