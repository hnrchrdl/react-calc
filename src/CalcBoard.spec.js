import React from 'react';
import renderer from 'react-test-renderer';
import CalcBoard from './CalcBoard';
import store from './store';
import { Provider } from 'react-redux';

it('matches snapshots without props', () => {
  const component = renderer.create(
    <Provider store={store}>
      <CalcBoard />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
it('matches snapshots with results', () => {
  // render results
  const component = renderer.create(
    <Provider store={store}>
      <CalcBoard results={'1.1111'} steps={['1', '+', '0.1111']} />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
// render steps
it('matches snapshots with results', () => {
  const component = renderer.create(
    <Provider store={store}>
      <CalcBoard results={null} steps={['1', '+', '0.1111']} />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
