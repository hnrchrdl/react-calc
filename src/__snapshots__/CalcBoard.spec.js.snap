// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots with results 1`] = `
<div
  className="calc-board"
>
  <div
    className="display"
  />
  <div
    className="calc-board-container"
  >
    <div
      className="calc-board-rows"
    >
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          1
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          2
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          3
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          4
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          5
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          6
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          7
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          8
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          9
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          0
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          .
        </div>
      </div>
    </div>
    <div
      className="calc-board-rows"
    >
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          +
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          -
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          *
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          /
        </div>
      </div>
    </div>
    <div
      className="calc-board-rows"
    >
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          C
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          CC
        </div>
      </div>
    </div>
    <div
      className="calc-board-rows"
    >
      <div
        className="pad"
        onClick={[Function]}
      >
        =
      </div>
    </div>
  </div>
</div>
`;

exports[`matches snapshots with results 2`] = `
<div
  className="calc-board"
>
  <div
    className="display"
  />
  <div
    className="calc-board-container"
  >
    <div
      className="calc-board-rows"
    >
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          1
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          2
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          3
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          4
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          5
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          6
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          7
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          8
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          9
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          0
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          .
        </div>
      </div>
    </div>
    <div
      className="calc-board-rows"
    >
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          +
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          -
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          *
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          /
        </div>
      </div>
    </div>
    <div
      className="calc-board-rows"
    >
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          C
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          CC
        </div>
      </div>
    </div>
    <div
      className="calc-board-rows"
    >
      <div
        className="pad"
        onClick={[Function]}
      >
        =
      </div>
    </div>
  </div>
</div>
`;

exports[`matches snapshots without props 1`] = `
<div
  className="calc-board"
>
  <div
    className="display"
  />
  <div
    className="calc-board-container"
  >
    <div
      className="calc-board-rows"
    >
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          1
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          2
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          3
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          4
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          5
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          6
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          7
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          8
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          9
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          0
        </div>
        <div
          className="pad"
          onClick={[Function]}
        >
          .
        </div>
      </div>
    </div>
    <div
      className="calc-board-rows"
    >
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          +
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          -
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          *
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          /
        </div>
      </div>
    </div>
    <div
      className="calc-board-rows"
    >
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          C
        </div>
      </div>
      <div
        className="calc-board-row"
      >
        <div
          className="pad"
          onClick={[Function]}
        >
          CC
        </div>
      </div>
    </div>
    <div
      className="calc-board-rows"
    >
      <div
        className="pad"
        onClick={[Function]}
      >
        =
      </div>
    </div>
  </div>
</div>
`;
